### DISCUSSION TITLE
Are shared component js chunks documented anywhere?

### DISCUSSION THREAD

**Discussion by runmoore** [remix-run/remix#2295] [Q&A] [answered]
Is this technique documented anywhere in the docs? I tried looking but couldn't find anything that mentions this specifically.

### What do I mean by **shared component js chunks**?

If you have a React component called `SharedComponent` that is used in multiple pages. Remix extracts the component into it's own js bundle and ships it separately to the browser. The manifest updates to reference it, and the individual page chunks now no longer include the contents of this SharedComponent, instead they reference the shared chunk.

See image below

![Screenshot 2022-03-11 at 17 40 59](https://user-images.githubusercontent.com/1532576/157921876-c27d8aa7-6228-4172-9483-5e9120eb7509.png)

For complex components this has obvious benefits on reducing the amount of js transferred and I really like it but I would like to learn a bit more about it and couldn't find any reference to it in the docs.

**Comment by kiliman** [comment, marked as answer]
The code-splitting is handled by `esbuild`. https://esbuild.github.io/api/#splitting
